File: ad_library/src/main/java/com/adgrowth/internal/helpers/JSONHelper.java
package com.adgrowth.internal.helpers;

import androidx.annotation.Nullable;

import org.json.JSONException;
import org.json.JSONObject;

public class JSONHelper {
    public static String getString(JSONObject json, String field, String defaultValue) {
        try {
            String media_url = json.getString(field);
            if (media_url != null) return media_url;
            return defaultValue;
        } catch (JSONException e) {
            return defaultValue;
        }
    }

    public static String safeGetString(JSONObject json, String field, String defaultValue) {
        return getString(json, field, defaultValue);
    }

    public static String safeGetString(JSONObject json, String field) {
        return getString(json, field, "");
    }

    public static JSONObject safeGetObject(JSONObject json, String field) {
        return getObject(json, field, new JSONObject());
    }

    public static JSONObject safeGetObject(JSONObject json, String field, JSONObject defaultValue) {
        return getObject(json, field, defaultValue);
    }

    public static JSONObject getObject(JSONObject json, String field, JSONObject defaultValue) {
        try {
            return json.getJSONObject(field);
        } catch (JSONException e) {
            return defaultValue;
        }
    }

    public static Integer getInt(JSONObject json, String field, Integer defaultValue) {
        try {
            return json.getInt(field);
        } catch (JSONException e) {
            return defaultValue;
        }
    }


    public static Integer safeGetInt(JSONObject json, String field, Integer defaultValue) {
        return getInt(json, field, defaultValue);

    }

    public static Integer safeGetInt(JSONObject json, String field) {
        return getInt(json, field, 0);
    }

    public static double getDouble(JSONObject json, String field, double defaultValue) {
        try {
            return json.getDouble(field);
        } catch (JSONException e) {
            return defaultValue;
        }
    }

    public static double safeGetDouble(JSONObject json, String field, double defaultValue) {
        return getDouble(json, field, defaultValue);
    }

    public static double safeGetDouble(JSONObject json, String field) {
        return getDouble(json, field, 0.0);
    }


}

File: ad_library/src/main/kotlin/com/adgrowth/internal/integrations/adserver/helpers/JSONHelper.kt
package com.adgrowth.internal.integrations.adserver.helpers

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

object JSONHelper {
    private fun getString(json: JSONObject, field: String, defaultValue: String?): String? {
        return try {
            val mediaUrl = json.getString(field)
            mediaUrl ?: defaultValue
        } catch (e: JSONException) {
            defaultValue
        }
    }

    fun safeGetString(json: JSONObject, field: String, defaultValue: String?): String? {
        return getString(json, field, defaultValue)
    }

    fun safeGetString(json: JSONObject, field: String): String {
        return getString(json, field, "")!!
    }

    fun safeGetObject(json: JSONObject, field: String): JSONObject {
        return getObject(json, field, JSONObject())
    }

    fun safeGetBoolean(json: JSONObject, field: String, defaultValue: Boolean): Boolean {
        return getBoolean(json, field, defaultValue)
    }

    fun safeGetObject(json: JSONObject, field: String, defaultValue: JSONObject): JSONObject {
        return getObject(json, field, defaultValue)
    }

    fun safeGetArray(json: JSONObject, field: String): JSONArray {
        return getArray(json, field, JSONArray())
    }

    fun safeGetArray(json: JSONObject, field: String, defaultValue: JSONArray): JSONArray {
        return getArray(json, field, defaultValue)
    }

    private fun getObject(json: JSONObject, field: String, defaultValue: JSONObject): JSONObject {
        return try {
            json.getJSONObject(field)
        } catch (e: JSONException) {
            defaultValue
        }
    }

    private fun getBoolean(json: JSONObject, field: String, defaultValue: Boolean): Boolean {
        return try {
            json.getBoolean(field)
        } catch (e: JSONException) {
            defaultValue
        }
    }

    private fun getArray(json: JSONObject, field: String, defaultValue: JSONArray): JSONArray {
        return try {
            json.getJSONArray(field)
        } catch (e: JSONException) {
            defaultValue
        }
    }

    private fun getInt(json: JSONObject, field: String?, defaultValue: Int?): Int? {
        return try {
            json.getInt(field)
        } catch (e: JSONException) {
            defaultValue
        }
    }

    fun safeGetInt(json: JSONObject, field: String?, defaultValue: Int?): Int? {
        return getInt(json, field, defaultValue)
    }

    fun safeGetInt(json: JSONObject, field: String?): Int? {
        return getInt(json, field, 0)
    }

    private fun getDouble(json: JSONObject, field: String?, defaultValue: Double): Double {
        return try {
            json.getDouble(field)
        } catch (e: JSONException) {
            defaultValue
        }
    }

    fun safeGetDouble(json: JSONObject, field: String?, defaultValue: Double): Double {
        return getDouble(json, field, defaultValue)
    }

    fun safeGetDouble(json: JSONObject, field: String?): Double {
        return getDouble(json, field, 0.0)
    }
}
